add a signer claim to the aws alb auth provider

freben opened the pull request:
Adds an optional `signer` claim matcher to the configuration options of the AWS ALB auth provider.

This also correctly marks the `iss` claim as required in the config schema, which it always was in practice at runtime.

Rugvip reviewed (approved):
:ok_hand: 

Do you think we should make this required in the future? Thinking we could add a warning or smth if it's not set. Perhaps also with a somewhat stronger encouragement in the changeset of using the new option?

freben commented:
Changeset strengthened and added config instructions, updated api report.

Yeah I would love to make it required - but when? Now immediately and make it a breaking change, or give it some time? In all honesty, making this one required and the issuer optional would be a decent end state.

Rugvip commented:
Yep probably, but yep not now. I think the current change is good, and then we can move towards it being required in the future
